### PR TITLE
[PATCH v5] validation: random: limit duration of some tests

### PR DESCRIPTION
**Pull request by JereLeppanen**

### CODE REVIEW

File: test/validation/api/random/random.c [hunk 1]
```diff
@@ -230,9 +230,20 @@ static void random_test_frequency(odp_random_kind_t kind)
 		8929.74, 17419.3, 34224.0, 67587.1,
 	};
 
+	int bits_max = 8;
+
+#ifdef __ARM_FEATURE_RNG
```
Comment (MatiasElo) [inline test/validation/api/random/random.c:235]: I'm not that fond of adding architecture specific flags into the validation tests. Perhaps the same result could be achieved by measuring time and capping the test length to some maximum value?

Comment (JereLeppanen) [reply test/validation/api/random/random.c]: What I don't like about that is that you end up doing time measurements even when they're not needed, which is in most environments. But in practice it doesn't matter. Done in v2.

File: test/validation/api/random/random.c [hunk 2]
```diff
@@ -252,6 +254,9 @@ static void random_test_frequency(odp_random_kind_t kind)
 		       bits, chisq, cells - 1, crit, res_str(chisq < crit));
 
 		CU_ASSERT(chisq < crit);
+
+		if (odp_time_local_ns() - start > 5 * ODP_TIME_SEC_IN_NS)
```
[the remark this answers is not in the document]
Comment (JereLeppanen) [reply test/validation/api/random/random.c:258]: In some environments, with a one second limit, we break after executing the loop just once, and that's with bits=1, which is a kind of a special case. But I don't think that matters much. In the other place one second is probably fine.

[the remark this answers is not in the document]
Comment (MatiasElo) [reply test/validation/api/random/random.c:258]: One could check `if (bits > 1 && odp_time_local_ns() - start > ODP_TIME_SEC_IN_NS)`, but I'm fine with this version. The check can be optimized later if deemed necessary.

**Review by MatiasElo — APPROVED** [review]

File: test/validation/api/random/random.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (JereLeppanen) [reply test/validation/api/random/random.c]: Time limit is one second in v3, because there's not enough reasons for a longer limit.

**Comment by JereLeppanen** [comment]
v3:
- Add three new commits.

File: test/performance/odp_random.c [hunk 3]
```diff
@@ -322,25 +332,70 @@ static int test_random_latency(void *ptr)
...
+		}
+
+		rounds = (uint64_t)tr * options.msec * 1000000 / time;
+		free(buf);
 	}
+
+	if (rounds < 1)
+		rounds = 1;
+
+	return rounds;
```
Comment (MatiasElo) [inline test/performance/odp_random.c:385]: Could be one-liner.

File: test/performance/odp_random.c [hunk 4]
```diff
@@ -322,25 +332,70 @@ static int test_random_latency(void *ptr)
...
+		uint32_t r = 1, tr = 0;
+		uint64_t time, start = odp_time_local_ns();
+
+		while (1) {
+			if (options.mode)
+				random_data_latency(global, 0, r, buf, options.size);
+			else
+				random_data_loop(type, r, buf, options.size);
+
+			time = odp_time_local_ns() - start;
```
[the remark this answers is not in the document]
Comment (JereLeppanen) [reply test/performance/odp_random.c]: Using nsec directly makes for slightly clearer code, which is why I liked it better. But changed anyway in v4. Either way is fine with me,

**Comment by JereLeppanen** [comment]
v4:
- Matias' comments.

**Review by MatiasElo — APPROVED** [review]

**Comment by JereLeppanen** [comment]
v5:
- Rebase.
- Add review tags.

**Pull request merged**